onnxruntime/core/providers/nnapi/nnapi_builtin/builders/helper.h

```diff
@@ -172,5 +172,9 @@ inline uint32_t ShapeSize(const Shape& shape) {
 bool CheckIsInitializer(const InitializedTensorSet& initializers, const NodeUnit& node_unit,
                         const std::string& input_name, const char* input_description);
 
+// Convert ONNX int64 input to NNAPI int32 type input
+// Mostly used in handling `axes` input for now (including the handling for negative axis)
+std::vector<int32_t> OnnxShapeToNnapi(gsl::span<const int64_t> onnx_shape, std::optional<size_t> input_size);
```

Comment:
if this is mainly for axes, how about `OnnxAxesToNnapi`?
```suggestion
std::vector<int32_t> OnnxAxesToNnapi(gsl::span<const int64_t> onnx_axes, std::optional<size_t> input_rank);
```

Reply:
There's two aspects. One is the type conversion, second is handling negative axes.

Should we have `OnnxShapeToNnapi` only do the type conversion, and add `OnnxAxesToNnapi` which calls `OnnxShapeToNnapi` and does the negative axis handling after that?

Reply:
I'm being a bit picky about naming here. "shape" and "axes" have different meanings, maybe we can use a more general term like "indexes"? maybe then a single function is fine, but I'm also good with two.

Reply:
I think the current one single function setup looks good to me, and we have the option to handle negative axes if needed which is nice. otherwise may be a little too detailed if we split a simple couple lines helper into two, and probably would have to iterate twice through the vector 1. narrow the type 2. iterate again to handle the negative axis.

Agree we can have a more general naming for the method here. also considering in the future it's possible we might have other similar cases to handle other than "axes" when supporting more operators for nnapi.

Reply:
Let's go with Ed's suggestion for the single function.